### PR TITLE
Add new error handlers, add tests for NonConvergingErrorHandler

### PR DESCRIPTION
**Pull request by esoteric-ephemera**
Add error handling for
- PDSYEVX (analogous to PSSYEVX), possibly reflecting updated LAPACK routines in vasp 6
- PRICELV, similar error handling as POSMAP, see [VASP forum discussion](https://www.vasp.at/forum/viewtopic.php?p=25608)
- PSMAXN when a calculation doesn't converge electronically or ionically. I see there was earlier discussion of PSMAXN in [#133](https://github.com/materialsproject/custodian/issues/133), but handling was dropped from [#134](https://github.com/materialsproject/custodian/pull/134)
- Tests for the above, and `NonConvergingErrorHandler` to close [#300](https://github.com/materialsproject/custodian/issues/300)
- Replaced all POTCARs with fake ones (header info is the same, body is replaced with random numbers) using `pymatgen.dev_scripts.potcar_scrambler.py`
- All test POTCARs that permit gzipped i/o are now gzipped

### CODE REVIEW

File: tests/files/nonconv/POTCAR
Comment (janosh) [inline tests/files/nonconv/POTCAR:1]: maybe gzip this?

File: tests/vasp/test_handlers.py [hunk 1]
```diff
@@ -1046,3 +1065,71 @@ def test_correct(self):
         incar = Incar.from_file("INCAR")
         assert incar.get("PREC") == "High"
         assert incar.get("ENAUG", 0) == incar.get("ENCUT", 2) * 2
+
+
+class NonConvergingErrorHandlerTest(PymatgenTest):
```
Comment (janosh) [inline tests/vasp/test_handlers.py:1070]: yay! great to finally have test coverage for `NonConvergingErrorHandler`!

File: tests/vasp/test_handlers.py [hunk 2]
```diff
@@ -427,29 +429,33 @@ def test_bravais(self):
...
-        incar = Incar.from_file("INCAR")
-        assert incar["SYMPREC"] == pytest.approx(1e-6)
-
-        assert handler.check() is True
-        assert handler.correct()["errors"] == ["posmap"]
-        incar = Incar.from_file("INCAR")
-        assert incar["SYMPREC"] == pytest.approx(1e-4)
+    def test_posmap_and_pricelv(self) -> None:
+        incar_orig = Incar.from_file("INCAR")
+        # Joining tests for these three tags as they have identical handling
```
Comment (janosh) [inline tests/vasp/test_handlers.py:434]: good comment!

File: custodian/vasp/handlers.py [hunk 3]
```diff
@@ -564,18 +569,24 @@ def correct(self):
             else:
                 actions.append({"dict": "INCAR", "action": {"_set": {"ISYM": 0}}})
 
-        if "posmap" in self.errors:
+        if symprec_errors := self.errors & {"posmap", "pricelv"}:
```
Comment (janosh) [inline custodian/vasp/handlers.py:572]: some fancy code here 👍 😄

**Review by janosh — CHANGES_REQUESTED** [review]
great stuff, esp. all the new tests! thanks @esoteric-ephemera! 👍

File: tests/files/nonconv/POTCAR
Comment (esoteric-ephemera) [reply tests/files/nonconv/POTCAR:1]: It's not needed for the tests, so I removed it. Think it's worth it to switch out the other POTCARs here for fake ones like in PMG?

File: custodian/vasp/handlers.py (hunk 3 above)
Comment (esoteric-ephemera) [reply custodian/vasp/handlers.py:572]: I learned from the best!

File: tests/files/nonconv/POTCAR
Comment (janosh) [reply tests/files/nonconv/POTCAR:1]: i think so. maybe not all of them are needed in the 1st place

Comment (esoteric-ephemera) [reply tests/files/nonconv/POTCAR:1]: Done - last thing is to figure out why the `pssyevx` test fails in CI but passes locally

Comment (esoteric-ephemera) [reply tests/files/nonconv/POTCAR:1]: Temporarily unzipped these so the tests pass. Just seems like the `*.gz` files aren't being properly found by pytest. Maybe a config thing?

**Review by janosh — APPROVED** [review]
yeah buddy! this repo feels light weight now with all those compressed POTCARs 😄

**Pull request merged**